images/router/haproxy/conf/haproxy-config.template

```diff
@@ -439,6 +442,15 @@ backend {{genBackendNamePrefix $cfg.TLSTermination}}:{{$cfgIdx}}
   http-request add-header Forwarded for=%[src];host=%[req.hdr(host)];proto=%[req.hdr(X-Forwarded-Proto)]
   {{- end }}
 
+  {{- with $pathRewriteTarget := firstMatch $pathRewriteTargetPattern (index $cfg.Annotations "haproxy.router.openshift.io/rewrite-target") }}
+  # Path rewrite target
+    {{- if eq $pathRewriteTarget "/" }}
+  http-request replace-path ^{{ $cfg.Path }}/?(.*)$ {{ $pathRewriteTarget }}\1
```

Comment:
The pattern looks suspicious.  Suppose `$cfg.Path` were `/foo`, `$pathRewriteTarget` were `/bar`, and a client requested `/foobar`; would this rule rewrite the request to `/barbar`?  I would expect the pattern to be something like this: `^{{ $cfg.Path }}(?:/(.*)|$)$`.  Similar applies to the pattern below.

Reply:
> `^{{ $cfg.Path }}(?:/(.*)|$)$`

Could probably be simplified to the following: `^{{ $cfg.Path }}(?:/(.*))?$`.

Reply:
@Miciah about your first remark, if user has `$pathRewriteTarget` set to `/bar` it doesn't go in this if cause but in the else cause

Besides, in your example if client request `/foobar` it will not even reach the backend so we will not use at all this stuff.

it's because `os_http_be.map` file will contain something like
``^mydomain\.com(:[0-9]+)?/foo(/.*)?$     be_http:default:example`

Reply:
This is surprising to me as i would have expected the behaviour described by @Miciah to be true ;
Looking at the documentation provided by HAproxy regarding this annotation, one could expect that `/foobar` be replaced with `/barbar` when the annotation is set to `/bar` and the ingress path is `/foo` . (ref. line 5 in the table https://haproxy-ingress.github.io/docs/configuration/keys/#rewrite-target)  

Ingress path | Request path | Rewrite target | Output
/foo | /foobar | /bar | /barbar

I wouldn't expect `/` to be treated differently in the frontend as it's part of the "path" string...  

Now, I'm just giving my opinion from a user perspective... please don't mind it if this is something that should be addressed in the haproxy project... (I wouldn't know where to go to refer this as a potential design issue...)

Reply:
@elafontaine : well if you specify route path being `/foobar` then you can play with annotation on it but if host of the route is only `/foo` then you can only play with `/foo` and `/foo/something`
another app could register a backend service with `/foobar` host

about `/`being handled differently, it's to act like nginx, haproxy and traefik rewrite-target annotations.

Reply:
so I wouldn't be able to have `/foobar` still be redirected to `/barbar` because it wouldn't even be catched by the ingress resource  as being part of `/foo` ingress path.  Is my understanding right?

Reply:
(it would make sense when you think of them as directory/file endpoint, but that's not how I was seeing it before :) )